migrator/migrations/m_112_to_m_113_recreate_groups_bucket/migration_test.go

```diff
@@ -0,0 +1,168 @@
+	migratedGroup := &storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "io.stackrox.authz.group.migrated." + uuid.NewV4().String(),
+			AuthProviderId: "some-value",
+		},
+		RoleName: "some-value",
+	}
+	rawMigratedGroup, err := proto.Marshal(migratedGroup)
+	suite.NoError(err)
+
+	// invalidGroup should be dropped during re-creation.
+	invalidGroup := &storage.Group{
+		Props: &storage.GroupProperties{
+			Key: "some-value",
+		},
+		RoleName: "",
+	}
+	rawInvalidGroup, err := proto.Marshal(invalidGroup)
+	suite.NoError(err)
+
+	// The following cases represent the entries within the groups bucket _before_ migration.
+	// After migration, note that:
+	// - existing-group should not have been dropped, due to having an ID.
+	// - migrated-group should not have been dropped, due to having an ID.
+	// - invalid-group should have been dropped, due to no ID.
+	// - invalid-bytes should have been dropped, due to some weird data and no group proto message.
+	cases := map[string]struct {
+		entry                groupEntry
+		existsAfterMigration bool
+	}{
+		"existing-group": {
+			entry: groupEntry{
+				key:   []byte(existingGroup.GetProps().GetId()),
+				value: rawExistingGroup,
+			},
+			existsAfterMigration: true,
+		},
+		"migrated-group": {
+			entry: groupEntry{
+				key:   []byte(migratedGroup.GetProps().GetId()),
+				value: rawMigratedGroup,
+			},
+			existsAfterMigration: true,
+		},
+		"invalid-group": {
+			entry: groupEntry{
+				key:   rawInvalidGroup,
+				value: []byte(""),
+			},
+		},
+		"invalid-group-stored-by-id": {
+			entry: groupEntry{
+				key:   []byte(existingGroup.GetProps().GetId() + "make-it-unique"),
+				value: rawInvalidGroup,
+			},
+		},
+		"invalid-bytes": {
+			entry: groupEntry{
+				key:   []byte("some-random-bytes-no-one-knows"),
+				value: []byte("some-other-random-bytes"),
+			},
+		},
+	}
+
+	var expectedEntriesAfterMigration int
+	for _, c := range cases {
+		if c.existsAfterMigration {
+			expectedEntriesAfterMigration++
+		}
+	}
+
+	// 1. Migration should succeed if the bucket does not exist.
+	suite.NoError(recreateGroupsBucket(suite.db))
+
+	// 2. Add the groups to the groups bucket before running the migration.
+	err = suite.db.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucketIfNotExists(bucketName)
+		suite.NoError(err)
+		for _, c := range cases {
+			suite.NoError(bucket.Put(c.entry.key, c.entry.value))
+		}
+		return nil
+	})
+	suite.NoError(err)
+
+	// 3. Run the migration to re-create the groups bucket and remove invalid entries.
+	suite.NoError(recreateGroupsBucket(suite.db))
+
+	// 4. Verify that all entries are as expected.
+	err = suite.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(bucketName)
+
+		for _, c := range cases {
+			// In case the entry should not exist, it shouldn't be possible to retrieve any values from the given key.
+			if !c.existsAfterMigration {
+				suite.Empty(bucket.Get(c.entry.key))
+				continue
```

Comment:
super nit: why not `else`?

Reply:
No particular reason, changed it to use `else`.